### PR TITLE
docs(core): improve caching feature page

### PR DESCRIPTION
**Pull request by juristr**
https://nx-dev-git-fork-juristr-docs-fp-cache-task-results-nrwl.vercel.app/features/cache-task-results

### CODE REVIEW

File: docs/shared/features/cache-task-results.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # Cache Task Results
 
-It's costly to rebuild and retest the same code over and over again. Nx has the most sophisticated and battle-tested computation caching system to make sure it never rebuilds the same code twice. It knows when the task you are about to run, has been executed before, so it can use the cache to restore the results of running that task.
+Rebuilding and retesting the same code repeatedly is costly. Nx offers a sophisticated and battle-tested computation caching system that ensures **code is never rebuilt twice**. This:
 
-If you want to learn more about the conceptual model behind Nx's caching, read [How Caching Works](/concepts/how-caching-works).
+- drastically **speeds up your task execution times** while developing locally, particularly [on CI](/ci/features/remote-cache)
```
Comment (isaacplmann) [inline docs/shared/features/cache-task-results.md:5]:
```suggestion
- drastically **speeds up your task execution times** while developing locally and even more [in CI](/ci/features/remote-cache)
```

File: docs/shared/features/cache-task-results.md [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+- **saves you money on CI/CD costs** by reducing the number of tasks that need to be executed
+
+Nx **restores the terminal output, along with the files and artifacts** created from running the task (e.g., your build or dist directory). If you want to learn more about the conceptual model behind Nx's caching, read [How Caching Works](/concepts/how-caching-works).
```
Comment (isaacplmann) [inline docs/shared/features/cache-task-results.md:8]:
```suggestion
Nx **restores both the terminal output and the files** created from running the task (e.g., your build or dist directory). If you want to learn more about the conceptual model behind Nx's caching, read [How Caching Works](/concepts/how-caching-works).
```

File: docs/shared/features/cache-task-results.md [hunk 3]
```diff
@@ -51,33 +54,28 @@ the same output. As an example, e2e test runs that hit the backend API cannot be
 the result of the test run.
 {% /callout %}
 
-Now, if you run a `build` task twice, the second time the operation will be instant because it is restored from the cache.
-
-{% terminal-video src="/documentation/shared/images/caching/cache-terminal-animation.mp4" alt="Video showing the terminal output of running a build command first without cache and then with cache. The 2nd run is almost instant, taking just 18ms" /%}
+## Enable Remote Caching
 
-Nx restores both
+By default, Nx caches task results locally. Most of the benefits from caching come on CI, where you can **share the cache between different runs**. To enable remote caching connect your workspace to [Nx Cloud](/nx-cloud) by running the following command:
```
Comment (isaacplmann) [inline docs/shared/features/cache-task-results.md:59]:
```suggestion
By default, Nx caches task results locally. The biggest benefit of caching comes from using remote caching in CI, where you can **share the cache between different runs**. To enable remote caching, connect your workspace to [Nx Cloud](/nx-cloud) by running the following command:
```

File: docs/shared/features/cache-task-results.md [hunk 4]
```diff
@@ -111,39 +109,62 @@ To achieve this, we can add an `inputs` and `outputs` definition globally for al
...
 {% /tab %}
 {% /tabs %}
 
-Note, you only need to define output locations if they differ from the usual `dist` or `build` directory which Nx automatically recognizes.
+Note that you only need to define output locations if they differ from the usual `dist` or `build` directory, which Nx automatically recognizes.
 
 Learn more [about configuring inputs including `namedInputs`](/recipes/running-tasks/configure-inputs).
 
-## Where is the Cache Stored?
+## Configure caching automatically
```
Comment (isaacplmann) [inline docs/shared/features/cache-task-results.md:139]:
```suggestion
## Configure Caching Automatically
```

File: docs/shared/features/cache-task-results.md [hunk 5]
```diff
@@ -111,39 +109,62 @@ To achieve this, we can add an `inputs` and `outputs` definition globally for al
...
 
-The cache is stored in `.nx/cache` by default. You can also [change where the cache](/recipes/running-tasks/change-cache-location) is stored if you want.
+When using [Nx plugins](/concepts/nx-plugins), many caching configurations are handled automatically, saving you the effort of manual setup. **Nx plugins can [automatically infer tasks](/concepts/inferred-tasks) and configure caching** based on your underlying tooling configuration files.
```
Comment (isaacplmann) [inline docs/shared/features/cache-task-results.md:141]:
```suggestion
When using [Nx plugins](/concepts/nx-plugins), many tasks have caching configured automatically, saving you the effort of manual setup. **Nx plugins can [automatically infer tasks](/concepts/inferred-tasks) and configure caching** based on your underlying tooling configuration files.
```

File: docs/shared/features/cache-task-results.md [hunk 6]
```diff
@@ -111,39 +109,62 @@ To achieve this, we can add an `inputs` and `outputs` definition globally for al
...
-You can enable remote caching (Nx Replay) by connecting to [Nx Cloud](/ci/features/remote-cache). To connect Nx to Nx Cloud, [create an account on cloud.nx.app](https://cloud.nx.app) and connect to your repository.
+```shell
+npx nx add @nx/vite
+```
 
-Learn more about [remote caching](/ci/features/remote-cache).
+...it automatically detects your `vite.config.ts` file, infers the tasks you'd be able to run, such as `build`, and **automatically configures the cache settings** for these tasks as well as the [task pipeline](/concepts/task-pipeline-configuration) (e.g., triggering dependent builds).
 
-## Turn off or Skip the Cache
+This means **you don't need to manually specify cacheable operations for Vite tasks** and the cache setting such as input and outputs are always in sync with the `vite.config.ts` file.
```
Comment (isaacplmann) [inline docs/shared/features/cache-task-results.md:151]:
```suggestion
This means **you don't need to manually specify cacheable operations for Vite tasks** and the cache setting such as inputs and outputs are always in sync with the `vite.config.ts` file.
```

File: docs/shared/features/cache-task-results.md [hunk 7]
```diff
@@ -111,39 +109,62 @@ To achieve this, we can add an `inputs` and `outputs` definition globally for al
...
 ```
 
-Alternatively if you want to disable caching for a particular task, just make sure it is not part [of the cached tasks](/features/cache-task-results#define-cacheable-tasks). If [you're using Nx Cloud](/ci/features/remote-cache#skipping-cloud-cache), you might want to use `--no-cloud` to skip remote caching.
+Alternatively, you can view these directly in your editor by installing [Nx Console](/getting-started/editor-setup).
 
-## Clear the Local Cache
+Learn more details about [Nx plugins](/concepts/nx-plugins) and [inferred tasks](/concepts/inferred-tasks).
 
-To clear the local cache, run:
+## Troubleshooting Caching
```
Comment (isaacplmann) [inline docs/shared/features/cache-task-results.md:163]:
```suggestion
## Troubleshoot Cache Settings
```

File: docs/shared/recipes/running-tasks/skipping-cache.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# Skip Task Caching
```
Comment (isaacplmann) [inline docs/shared/recipes/running-tasks/skipping-cache.md:1]: We should add a reference to `nx reset` on this page as well.

**Review by isaacplmann — CHANGES_REQUESTED** [review]
Reworded a few things

File: docs/shared/recipes/running-tasks/skipping-cache.md (hunk 8 above)
Comment (juristr) [reply docs/shared/recipes/running-tasks/skipping-cache.md:1]: added a section

**Review by isaacplmann — APPROVED** [review]

**Pull request merged**